### DISCUSSION TITLE
[10.x]Why Laravel Cache::lock() not ‘set nx’ but ‘setnx’

### DISCUSSION THREAD

**Discussion by haodudecao** [laravel/framework#49270] [Q&A]
<img width="1233" alt="image" src="https://github.com/laravel/framework/assets/34431537/38486d20-55ce-406e-b38f-eaea110e4516">
redis setnx is deprecated

**Comment by henzeb** [comment]
The cache lock was made before deprecation ? I suggest create a PR for this.
